slsim/Sources/SupernovaeCatalog/supernovae_sample.py

```diff
@@ -47,47 +84,44 @@ def supernovae_host_galaxy_offset(host_galaxy_catalog):
             )[0]
 
         # Calculate offsets [rad]
-        offsets.append(offset_ratios[i] * list(host_galaxy_catalog["angular_size"])[i])
+        offset = offset_ratios[i] * list(host_galaxy_catalog["angular_size"])[i]
+        offsets.append(offset)
 
-        # Generate random angle
-        position_angle.append(np.random.uniform(0, 360))
+        galaxy_angle = np.random.uniform(0, np.pi)
+        supernova_angle = np.random.uniform(0, 2 * np.pi)
+        position_angle_galaxy.append(galaxy_angle)
+        position_angle_supernovae.append(supernova_angle)
 
-        # Calculate projected eccentricities
-        temp_e1, temp_e2 = galaxy_projected_eccentricity(
-            host_galaxy_catalog["ellipticity"][i]
+        # Calculate the x and y coordinates of the offset [arcsec]
+        x_off = ((np.cos(supernova_angle * units.rad)) * (offset * units.rad)).to(
+            units.arcsec
         )
-        e1.append(temp_e1)
-        e2.append(temp_e2)
-
-    # Calculate the ra and dec coordinates of the offset [arcsec]
-    host_center = SkyCoord(1 * units.deg, 1 * units.deg, frame="icrs")
-    offsets = host_center.directional_offset_by(
-        position_angle * units.deg, offsets * units.rad
-    )
-    original_ra_off = (offsets.ra - 1 * units.deg).to(units.arcsec)
-    original_dec_off = (offsets.dec - 1 * units.deg).to(units.arcsec)
-
-    transformed_ra_off = []
-    transformed_dec_off = []
+        y_off = ((np.sin(supernova_angle * units.rad)) * (offset * units.rad)).to(
+            units.arcsec
+        )
+        original_x_off.append(x_off)
+        original_y_off.append(y_off)
 
-    # Transform the offset coordinates with eccentricities e1, e2 into elliptical coordinate system
-    for i in range(len(host_galaxy_catalog)):
+        # Calculate projected eccentricities
+        slsim_e1, slsim_e2 = galaxy_projected_eccentricity(
+            host_galaxy_catalog["ellipticity"][i], galaxy_angle * units.rad
+        )
+        e1.append(slsim_e1)
+        e2.append(slsim_e2)
 
-        # Conversion of slsim e1, e2 to lenstronomy e1, e2
-        lenstronomy_e1, lenstronomy_e2 = ellipticity_slsim_to_lenstronomy(e1[i], e2[i])
+        # Transform the offset coordinates with eccentricities e1, e2 into elliptical coordinate
+        # system
+        lens_e1, lens_e2 = ellipticity_slsim_to_lenstronomy(slsim_e1, slsim_e2)
+        lenstronomy_e1.append(lens_e1)
```

Comment:
Do you need to append lens_e1 and lens_e2? Because we use them intermediately and never used the full list anywhere.

Reply:
@sibirrer wanted to make sure that these values are recorded somewhere within the function

Reply:
@cmistick I don't think we need them. We can reproduce them from having the SLSim e1,e2 pairs